mne/decoding/receptive_field.py

```diff
@@ -0,0 +1,197 @@
+        _check_estimator(self.model)
+
+    def __repr__(self):  # noqa: D105
+        str_features = self.feature_names
+        s = "features : [%s, %s]" % (str_features[0], str_features[-1])
+        s += ", lags : [%f, %f]" % (self.lags[0], self.lags[-1])
+        s += ", model : %s" % str(self.model)
+        return "<ReceptiveField  |  %s>" % s
+
+    def fit(self, X, y):
+        """Fit a receptive field model.
+
+        Parameters
+        ----------
+        X : array, shape ([n_epochs], n_features, n_times)
+            The input features for the model.
+        y : array, shape ([n_epochs], n_times)
+            The output feature for the model.
+        """
+        n_feats = X.shape[1] if X.ndim == 3 else X.shape[0]
+        X_del, msk = self._delay_for_fit(X)
+
+        # Reshape so we have times x features
+        X_del = X_del[..., ~msk]
+        y = y[..., ~msk]
+
+        if X_del.ndim == 4:
+            # We have epochs, so hstack now
+            n_ep = X_del.shape[1]
+            X_del = X_del.swapaxes(0, 1).reshape(n_ep, -1, X_del.shape[-1])
+            X_del = np.hstack(X_del)
+            y = np.hstack(y)
+
+        # Update feature names if we have none
+        if self.feature_names is None:
+            self.feature_names = ['feature_%s' % ii for ii in range(n_feats)]
+        if len(self.feature_names) != n_feats:
+            raise ValueError('n_features in X does not match feature names '
+                             '(%s != %s)' % (n_feats, len(self.feature_names)))
+
+        # Reshape so time is last dimension
+        X_del = X_del.reshape([-1, X_del.shape[-1]])
+        y = y.reshape([-1, X_del.shape[-1]])
+
+        self.model.fit(X_del.T, y.T)
+        self.msk_remove = msk
+
+        coefs = get_coefs(self.model)
+        self.coef_ = coefs.reshape([len(self.feature_names), len(self.lags)])
+
+    def predict(self, X, y=None):
+        """Make predictions using a receptive field.
+
+        Parameters
+        ----------
+        X : array, shape ([n_epochs], n_channels, n_times)
+            The input features for the model.
+        y : None
+            Used for sklearn compatibility.
+
+        Returns
+        -------
+        y_pred : array, shape (n_times * n_epochs)
+            The output predictions with time concatenated.
+        """
+        X_del, msk = self._delay_for_fit(X)
+        X_del = X_del[..., ~msk]
+        X_del = X_del.reshape([-1, X_del.shape[-1]]).T
+        y_pred = self.model.predict(X_del)
+        self.mask_pred = ~msk
+        return y_pred
+
+    def _delay_for_fit(self, X):
+        # First delay
+        X_del = delay_time_series(X, self.lags, 1)
+
+        # Mask for removing edges later
+        msk_helper = np.ones(X.shape[-1])
+        msk_helper = delay_time_series(msk_helper, self.lags, 1)
+        msk = np.any(msk_helper == 0, axis=0)
+        return X_del, msk
+
+
+def delay_time_series(X, delays, sfreq=1., newaxis=0):
```

Comment:
not sure what you mean by `samples`. You mean instead of `delays`? Or do you mean to switch this to `tmin/tmax` as well?